pkg/controllers/nodeclaim/consistency/controller.go

```diff
@@ -95,18 +99,46 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim
 	if err != nil {
 		return reconcile.Result{}, nodeclaimutil.IgnoreDuplicateNodeError(nodeclaimutil.IgnoreNodeNotFoundError(err))
 	}
+	if err = c.checkConsistency(ctx, nodeClaim, node); err != nil {
+		return reconcile.Result{}, err
+	}
+	if !equality.Semantic.DeepEqual(stored, nodeClaim) {
+		// We call Update() here rather than Patch() because patching a list with a JSON merge patch
+		// can cause races due to the fact that it fully replaces the list on a change
+		// Here, we are updating the status condition list
+		if err = c.kubeClient.Status().Update(ctx, nodeClaim); err != nil {
+			if errors.IsConflict(err) {
+				return reconcile.Result{Requeue: true}, nil
+			}
+			return reconcile.Result{}, client.IgnoreNotFound(err)
```

Comment:
nit: any reason not to do the not found check in the initial error checking on L109?